onnxruntime/core/mlas/lib/platform.cpp

```diff
@@ -70,6 +82,10 @@ MLASCPUIDInfo::MLASCPUIDInfo()
 
     has_arm_neon_i8mm_ = ((getauxval(AT_HWCAP2) & HWCAP2_I8MM) != 0);
     has_arm_sve_i8mm_ = ((getauxval(AT_HWCAP2) & HWCAP2_SVEI8MM) != 0);
+
+    has_arm_sve_ = ((getauxval(AT_HWCAP) & HWCAP_SVE) != 0);
+    has_arm_neon_bf16_ = ((getauxval(AT_HWCAP2) & HWCAP2_BF16) != 0);
```

Comment:
how are these feature flag used? it seems to me that they are only used in test code? could you point me to the logic that the feature flag is used to determine which kernel to call during runtime?

Reply:
these flags are used in the following logic to detect whether bf16 feature is supported or not.

1. cpu provider matmul.h
https://github.com/microsoft/onnxruntime/pull/17031/files#diff-b64379f9113c5d88446bf978474921915149b31ca703038f06673b81cf552a48R33

```
#if defined(__aarch64__) && defined(__linux__)
    auto config_ops = info.GetConfigOptions().GetConfigEntry(kOrtSessionOptionsGemmFastMathMode);
    use_fastmath_mode_ = (config_ops == "1") && CPUIDInfo::GetCPUIDInfo().HasArmNeon_BF16();
#endif
```

2. and in sbgemm 
https://github.com/microsoft/onnxruntime/pull/17031/files#diff-a6732e6798dee7a36040e9c388882279bbb70f1e53372a0a751b149429346118

```
bool MLASCALL
MlasBf16AccelerationSupported()
{
#if defined(MLAS_TARGET_ARM64)
    return MLAS_CPUIDINFO::GetCPUIDInfo().HasArmNeon_BF16();
#else
    return false;
#endif
```

Reply:
What about sve_bf16? is it useful? Should it be HasArmNeon_BF16() || HasArmSVE_BF16() instead?  If here the feature we care about is bf16 support, does it make sense to provide a simpler interface like HasBf16()?

Reply:
Currently I have added only neon bf16 kernels, not SVE. so, i'm not using sve_bf16() flags yet. but defined them for future use.
regarding unifying them to single interface, it's a good suggestion. but given they are two different extensions and can be supported independently , I think it's better to check for each of them separately and implement kernels accordingly .

Reply:
Are neon-bf16 and sve-bf16 require different kernels? If that is the case, then maybe you need more changes to have kernel function pointer setup based on hardware feature detection like:
https://github.com/microsoft/onnxruntime/blob/f45ef1da184c3c74e8e935cdaf12b67dcd953ddb/onnxruntime/core/mlas/lib/platform.cpp#L540 ?

It would be great if you can add comments accordingly to explain purpose of this flag and future steps to use it. or just remove it.

Reply:
thanks. I will remove the SVE flags. I wouldn't expect a scenario where both neon and sve kernels are active at the same time. since SVE is a superset of scenarios neon covers, I would assume people will prefer SVE over neon and deprecate the older ones. so, the kernel function pointer will always be one, pointing to the highest performant one.